src/components/OfflineWithFeedback.tsx

```diff
@@ -111,7 +111,15 @@ function OfflineWithFeedback({
                 }
 
                 return React.cloneElement(child, props);
-            }),
+            });
+
+            // If we only have one child, we want to return it as is, not as an array
```

Comment:
Actually, I think we just want to preserve the input structure...

If we're starting with an array of a single node, I think we'd like to output an array of a single node.

Reply:
> Actually, I think we just want to preserve the input structure...

In that case, if we take a look at what `Children.map` returns based on [React docs](https://react.dev/reference/react/Children#children-map-returns), then without this piece:

```jsx
            if (Array.isArray(strikedThroughChildren) && strikedThroughChildren.length === 1) {
                return strikedThroughChildren[0];
            }
```

We are not preserving initial structure (the `Children.map` returns an array even for single child (not counting falsey values).

Reply:
> This is a mess...

It indeed is :D That's why many thanks for diving that deep into it with me 👍 

> Coming back to Earth. This is not a clean situation, because we're doing dynamic dark magic over React component tree. We've already got a regression here, so we'd like to avoid another one.

> I'm wondering what's the likelihood of breaking another component's contract in the opposite way, i.e. providing a direct single child when children was typed as ReadonlyArray<ReactNode>. I guess it's low, but the unclean thing is that it cannot be verified with this dynamic approach.

Yes another regression, especially implicit one, is what we need to avoid the most. Implementing such helper over built-in mechanism might be a double-edged sword, as you've mentioned with potential reversed issue. 

I would say that we can take advantage of your helper just for that case - even if we use it once it still has an advantage in form of well-documented usage via JSDoc, separate place indicating its special function, instead of random inlining 👍 

This way we also avoid potential issues with other places where `Children.map` is used, as they stay intact.

I've plugged helper in code and it works properly, but in order for types to be compliant with `Children.map`, the return value was changed to `ReactElement` - thank you for explaining `SingleReactElement` above.

Let me know if you have anything else regarding that issue and once again thanks for thoughtful help 👍